src/main/java/org/wso2/carbon/inbound/custom/poll/MQConnectionBuilder.java

```
@@ -0,0 +1,243 @@
+                                }
+                            }
+                            if (isConnected) {
+                                logger.info("Queue Manager connected for " + hostport[0] + " " + hostport[1] + " " + channel);
+                                break;
+                            }
+                        }
+                        if (isConnected) {
+                            break;
+                        }
+                    }
+                }
+            }
+        } catch (MQException e) {
+            logger.info("Error initializing queue manager");
+        }
+        return this;
+    }
+
+    /**
+     * Initialize queue manager
+     */
+    public MQQueueManager getQueueManager(String host, String channel, int port) throws MQException {
+
+        MQEnvironment.hostname = host;
+        MQEnvironment.channel = channel;
+        MQEnvironment.port = port;
+
+        Future<String> control
+                = Executors.newSingleThreadExecutor().submit(new Callable<String>() {
+            public String call() throws Exception {
+                queueManager = new MQQueueManager(config.getqManger());
+                return "Initialized";
+            }
+        });
+
+        try {
+            control.get(2, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        } catch (TimeoutException e) {
+            logger.info("Connection timeout for " + host + " " + channel + " " + port);
+            control.cancel(true);
+            return null;
+        }
+        return queueManager;
+    }
+
+    /**
+     * Closing IBM-MQ connection
+     */
+    public void closeConnection() {
+        try {
+            if (queueManager.isConnected()) {
+                queueManager.close();
+                queueManager = null;
+            }
+        } catch (MQException e) {
+
```

Why this empty catch block?